File: qc-core/src/main/java/com/eager/questioncloud/core/domain/verification/repository/EmailVerificationRepository.java
package com.eager.questioncloud.core.domain.verification.repository;


import com.eager.questioncloud.core.domain.verification.model.EmailVerification;
import com.eager.questioncloud.core.domain.verification.vo.EmailVerificationType;

public interface EmailVerificationRepository {
    EmailVerification get(String token, EmailVerificationType emailVerificationType);

    EmailVerification getCreateUserVerification(Long userId);

    EmailVerification getForResend(String resendToken);

    EmailVerification save(EmailVerification emailVerification);
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/verification/infrastructure/repository/EmailVerificationRepository.kt
package com.eager.questioncloud.core.domain.verification.infrastructure.repository

import com.eager.questioncloud.core.domain.verification.enums.EmailVerificationType
import com.eager.questioncloud.core.domain.verification.model.EmailVerification


interface EmailVerificationRepository {
    fun get(token: String, emailVerificationType: EmailVerificationType): EmailVerification

    fun getCreateUserVerification(userId: Long): EmailVerification

    fun getForResend(resendToken: String): EmailVerification

    fun save(emailVerification: EmailVerification): EmailVerification

    fun update(emailVerification: EmailVerification)
}
